doc/how-to-source-packages.rst

```diff
@@ -0,0 +1,65 @@
+How to source packages
+======================
+
+Package Source
+--------------
+
+When we publish a package, its source is available from the package repository
+[#]_. The other way to access a package's source is from a source repository [#]_.
+
+Source Repositories
+-------------------
+
+Source repositories are a way to specify where to find the source code for a
+package, both for :ref:`package authors<pkg-authors>` and maintainers and for
+:ref:`package consumers<pkg-consumers>`.
+
+A relatively structured set of version control system (VCS) fields, that vary
+depending on the :ref:`VCS kind<vcs-kind>`, enables cabal commands and other
+tools to interpret and make effective use of this information.
+
+- ``source-repository`` says where to find the source for a **package description**
+- ``source-repository-package`` locates source for **project dependencies**
+
+.. _pkg-authors:
+
+As a Package Author
+-------------------
+
+If you are authoring or maintaining a package and want to link to the
+:ref:`package source<pkg-author-source>` you will be dealing with
+``source-repository``, a package description stanza that typically specifies
+where upstream development of the package is happening.  A package published to
+hackage will be displayed with a link to the source repository if a
+``source-repository`` stanza is present in the package description.
+
+.. _pkg-consumers:
+
+As a Package Consumer
+---------------------
+
+Cabal commands that work with dependencies actually need to have the source
+code of each dependency and will download it as needed. Example of commands
+like this are ``cabal build`` or ``cabal freeze``.
+
+Dependencies of packages in a project are sourced, by default, from the hackage
+package repository. Cabal will download the source for these from hackage.
+
+We can also take dependencies on :ref:`source packages<pkg-consume-source>`
+that are accessed via a version control system (VCS) such as git and set up in
+the project with ``source-package-repository`` stanzas. With these, we can take
+dependencies on packages not published to hackage, on revisions not yet
+published or on forks.  This is the easiest way to work with a fork, much
+easier than using a git submodule.  A dependency taken this way, effectively
+adds a local package to the project much like listing the package's source
+directory in the packages field, except that the source code is downloaded by
+cabal using the version control system.
+
+.. [#] Each package on hackage is available for download as a ``.tar.gz``
```

Comment:
Hackage (uppercase)

Reply:
Footnote moved to note and casing fixed.